### DISCUSSION TITLE
Contemporary Light on front and backface

### DISCUSSION THREAD

**Discussion by GiovanniSer** [cnr-isti-vclab/meshlab#1382] [General]
Hi all,
I am unable to find the option to have the double illumination (old lamp icon). It was very usefull because the gray back side of the face does not show the borders, so even if add the "WIREFRAME" mode with the gray faces, i am unable to view the differences between each back face.
I am usually working on shapes for 3D print, so I have always gray faces, on the back I cannot check the mesh.
BR
G.

**Comment by cignoni** [comment]
It is in the surface shading tab, you have three option for shading faces, **_Single_** (only front faces are shaded) , **_Double_** (front and back faces are shaded) and **_Fancy_** (front and back faces are shaded with slightly different color lights). 
See below:

<img width="400" alt="Screenshot 2023-04-30 at 11 39 43" src="https://user-images.githubusercontent.com/8055819/235346515-cc919188-c46b-4b4f-ab8b-f315354780e2.png">
<img width="400" alt="Screenshot 2023-04-30 at 11 39 40" src="https://user-images.githubusercontent.com/8055819/235346518-7645cda1-c78f-4469-879c-45690da49e0e.png">
<img width="400" alt="Screenshot 2023-04-30 at 11 39 35" src="https://user-images.githubusercontent.com/8055819/235346519-3870ce31-3f88-43f1-b5f8-121078785cd7.png">

**Comment by GiovanniSer** [comment]
Thanks, but I was hoping there was still the icon feature buried somewhere, because that's three steps against one.
I also have another little problem, now from some version, the settings window opens automatically when I enter.
And I always have to close it to have more space, because I don't need it.
Isn't there a way to create scripts that automatically perform a series of actions on command? It would be very interesting. Thanks again.
G.

**Reply by cignoni** [reply]
for scripting there is always [pymeshlab](https://pymeshlab.readthedocs.io/en/latest/)
